draft-ietf-radext-radiusdtls-bis.md

```diff
@@ -381,6 +381,19 @@ The above requirements are a logical extension of the common practice where a cl
 
 In an ideal world, a proxy could also apply the suggestion of the previous section, by discarding Acct-Delay-Time from Accounting-Request packets, and replacing it with Event-Timestamp.  However, this process is fragile and is not known to succeed in the general case.
 
+## Session limits and timeout
+
+While RADIUS/UDP could be implemented mostly stateless (except for the requests in flight), both TCP/TLS as well as DTLS require state tracking of the underlying TLS connection and are thus subject to potential resource exhaustion. This is aggravated by the fact that radius client/servers often from long-running peer relationships translating to long-running connections.
+
+Implementations SHOULD have configurable limits on the number of open connections. When this maximum is reached and a new session is started, the server MUST either drop an old session in order to open the new one or not create a new session. 
+
+The close notification of (D)TLS or underlying connections are not fully reliable, or they might be unnecessarily kept alive by hartbeat or watchdog traffic.
+Therefoe, both RADIUS/(D)TLS clients and servers SHOULD close connections after they have been idle for some time (no traffic except application layer watchdog). This idle timeout SHOULD be configurable within reasonable limits. 
```

Comment:
I can imagine that this can lead to problems.

One would be Reverse-CoA where you rely on sending the CoA packet "backwards" through RAIDUS/(D)TLS and that doesn't work if the client is behind a NAT, unless the connection stays open.

The second would be monitoring.
If you have a number of Service Providers, all behind a NAT, then the only good monitoring for Liveness of the SP is "does the server have a (D)TLS session with this client"
If the client would just close it, then this monitoring would not be possible anymore.

I would suggest: Replace `SHOULD` with `MAY` and add some text like `[...], the configuration SHOULD allow to disable the idle timeout session shutdown`

Reply:
maybe add a note about reverse CoA, and say also that systems SHOULD leave at least one connection open, to avoid delays opening new connections when traffic goes from zero to >0

Reply:
Have to get a bit more into Reverse-CoA.

I guess in the end this puts even more emphasis on the statement to which values to use:  'it very much depends...'. There is no one size fits all here.

Reply:
I think advising cilents to keep a connection open would be a bit misplaced in _this_ document (e.g. with dynamic discovery it might not be apropriate), but its very valid to acknowledge that some scenarios do require it.

A quick cross-read of reverse-CoA, it might more clearly state that the client MUST keep at least one conneciton open (it's kind of implied, but not clearly stated).